0035-feature-flags/README.md

```diff
@@ -0,0 +1,418 @@
+  # Since is the version a feature is first introduced in alpha stage.
+  Since
+  # Until is the version when a deprecated feature is fully removed. Historical versions included.
+  Until
+  # Stage describes what level of done-ness a feature is. TODO describe this better
+  Stage
+}
+```
+
+### API
+Below are descriptions of each function in the flag API and their intended UX.
+
+#### With()
+```
+pkg flag 
+...
+
+// With takes a context and a slice of one or many flags, setting the context on each. Duplicated flags will not error
+// and are treated as idempotent. If flags are duplicated with different fields, the value from the "latest" flag at the
+// tail slice will take precedence and merge over the prior field. The value of the field is not compared.
+// TODO Example [{name: "foo", version:  }]
+func With(ctx, []Flag) (context.Context, error) { ... }
+
+// DISCUSS: Instead of merging two flags and taking the latest, should we instead treat flags as a unique set, and error
+// if passed a duplicate flag?
+
+// TODO: Describe set-union expectations and how we'll error on duplicated feature name.
+```
+
+#### IsEnabled()
+```
+pkg flag 
+...
+
+// IsEnabled allows users to optimistically check a flag from ctx without regard for errors. Useful for control flow.
+func IsEnabled(ctx, flagName) bool { ... }
+```
+
+#### From()
+```
+pkg flag 
+...
+
+// From takes a ctx and the name of a flag, and returns it from the ctx object. If the doesn't exist, then it will
+// return an error.
+func From(ctx, flagName) (Flag, error) { ... }
+
+// alt, ok + empty flag on false? I feel like "ok" has gone out of style. dogsledding the flag itself isn't the worst UX
+// though. Kinda just expect Flag, error if we're returning a struct type.
+func From(ctx, flagName) (bool, Flag) { ... }
+
+// e.g.
+// if ok, _ := foo(); ok {
+//   ...		
+// }
+```
+
+#### All()
+```
+pkg flag 
+...
+
+// From takes a ctx and returns all flags from the ctx object.
+func All(ctx) ([]Flag)
+
+TODO alt
+func All(ctx) (map[flagName string]Flag)
+
+// NOTE: Getting back an empty collection here is not considered an error. The intended way to check for no flags is
+// if len(flags) == 0 {...}
+```
+
+### TODO DISCUSS: Global API
+- Enable globals? Potentially an abandoned idea.
+- Proposal:
+- Allow for global flags and have the API fallback to globals if no ctx is provided or ctx is empty?
+- Globals have advantages for developer ease of use in the SDK. e.g. get sensible defaults for loading up Zarf as a library, don't worry about enabling anything yourself.
+- Also if maintainers rely on enabling feature flags at specific versions then this keeps SDK to CLI parity.
+- Without global defaults we have to flip the boolean from enabled to disabled when providing a beta feature by default.
+- A major drawback with ctx is that it's difficult to inspect for users. "where did this flag come from?"
+- Maybe this is a good nudge to get features into GA sooner. More to discuss here.
```

Comment:
++ It is a great reason for us to have a process that tracks current state and what we require for graduation?